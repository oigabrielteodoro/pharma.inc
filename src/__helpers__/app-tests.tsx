import { ReactElement } from 'react'
import { render as rtlRender } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import nock from 'nock'

import { AppConfiguration } from 'config'

function render(ui: ReactElement) {
  return {
    ...rtlRender(ui, {
      wrapper: AppConfiguration,
    }),
  }
}

export * from '@testing-library/react'
export { render, userEvent, nock }
