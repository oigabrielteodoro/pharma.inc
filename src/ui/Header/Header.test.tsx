import React from 'react'

import { render, screen } from '__helpers__/app-tests'

import { Header } from '.'

describe('Header', () => {
  it('should be able render correctly', () => {
    render(<Header />)

    expect(
      screen.getByRole('heading', { name: /pharma inc/i }),
    ).toBeInTheDocument()
    expect(screen.getByLabelText(/user avatar/i)).toBeInTheDocument()
  })

  it('should be able snapshot is equal', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
