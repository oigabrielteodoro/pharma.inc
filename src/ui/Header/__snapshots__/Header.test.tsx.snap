// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able snapshot is equal 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background: #ffffff;
  box-shadow: 0 0 10rem 1rem rgba(0,0,0,0.1);
  padding: 1.6rem;
  z-index: 2;
}

.c1 {
  max-width: 110rem;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  color: #334155;
  font-size: 2.2rem;
}

.c2 span {
  color: #6366f1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 .userInfo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 1.6rem;
}

.c3 .userInfo strong {
  font-weight: 500;
  font-size: 1.4rem;
}

.c3 .userInfo small {
  margin-top: 0.4rem;
  color: #94a3b8;
  font-size: 1.2rem;
}

.c4 {
  color: #6366f1;
  border-radius: 0.5rem;
  background: #F5F3FF;
  width: 5rem;
  height: 5rem;
  display: grid;
  place-items: center;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        aria-label="pharma inc"
        class="c2"
      >
        Pharma 
        <span>
          Inc
        </span>
      </h1>
      <div
        class="c3"
      >
        <div
          class="userInfo"
        >
          <strong>
            Gabriel Teodoro
          </strong>
          <small>
            Front-End
          </small>
        </div>
        <div
          aria-label="user avatar"
          class="c4"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
        </div>
      </div>
    </div>
  </header>
</div>
`;
